skfda/exploratory/stats/_functional_transformers.py

```diff
@@ -342,3 +343,143 @@ def number_up_crossings(
+    data: Union[FDataBasis, FDataGrid],
+    n: int,
+) -> NDArrayFloat:
+    r"""
+    Calculate the specified unconditional moment of a dataset.
+
+    It performs the following map:
+    :math:`f(X)=\int_a^b f_1(X(t))dt,\dots,f_p(X(t))dt=\int_a^b  f^p(X(t))dt`.
+
+        Args:
+            data: FDataGrid or FDataBasis where we want to calculate
+            a particular unconditional moment.
+            n: order of the moment.
+
+        Returns:
+            ndarray of shape (n_dimensions, n_samples) with the values of the
+            specified moment.
+
+    Example:
+
+    We will calculate the first unconditional moment of the Canadian Weather
+    data set. In order to simplify the example, we will use only the first
+    five samples.
+    First we proceed to import the data set.
+    >>> from skfda.datasets import fetch_weather
+    >>> X = fetch_weather(return_X_y=True)[0]
+
+    Then we call the function with the samples that we want to consider and the
+    specified moment order.
+    >>> import numpy as np
+    >>> from skfda.exploratory.stats import unconditional_moments
+    >>> np.around(unconditional_moments(X[:5], 1), decimals=2)
+        array([[ 4.7 ,  4.03],
+               [ 6.16,  3.96],
+               [ 5.52,  4.01],
+               [ 6.82,  3.44],
+               [ 5.25,  3.29]])
+    """
+    return unconditional_expected_value(
+        data,
+        lambda x: pow(x, n),
+    )
+
+
+def unconditional_expected_value(
+    data: Union[FDataBasis, FDataGrid],
+    function: Callable[[np.ndarray], np.ndarray],
+) -> NDArrayFloat:
+    """
+    Calculate the unconditional expected value of a function.
+
+        Args:
+            data: FDataGrid or FDataBasis where we want to calculate
+            the expected value.
+            f: function that specifies how the expected value to is calculated.
+            It has to be a function of X(t).
+        Returns:
+            ndarray of shape (n_dimensions, n_samples) with the values of the
+            expectations.
+
+    Example:
+    We will use this funtion to calculate the logarithmic first moment
+    of the first 5 samples of the Berkeley Growth dataset.
+    We will start by importing it.
+    >>> from skfda.datasets import fetch_growth
+    >>> X = fetch_growth(return_X_y=True)[0]
+
+    We will define a function that calculates the inverse first moment.
+    >>> import numpy as np
+    >>> f = lambda x: pow(np.log(x), 1)
+
+    Then we call the function with the dataset and the function.
+    >>> from skfda.exploratory.stats import unconditional_expected_value
+    >>> np.around(unconditional_expected_value(X[:5], f), decimals=2)
+        array([[ 4.96],
+               [ 4.88],
+               [ 4.85],
+               [ 4.9 ],
+               [ 4.84]])
+    """
+    domain_range = data.domain_range[0][1] - data.domain_range[0][0]
+
+    if isinstance(data, FDataGrid):
+        return scipy.integrate.simpson(
+            function(data.data_matrix),
```

Comment:
You can apply the function directly to the `FDataGrid` and use its `integrate` method afterwards.

Reply:
Then the function should be a Callable[[FDataGrid], FDataGrid] so a FDataGrid is returned and the integrate method could be applied later

Reply:
You can apply a `ndarray` function over a `FDataGrid`. Mypy probably won't like it but it will work at runtime.